components/net/tests/local_directory_listing.rs

```diff
@@ -0,0 +1,740 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::collections::BTreeMap;
+
+use embedder_traits::resources::{read_string, Resource};
+use net::local_directory_listing::{
+    build_html_directory_listing, day_of_month_ordinal_suffix, write_html_safe, DirectoryItem,
+};
+use time_03::{Date, Month, OffsetDateTime, Time};
+
+enum FileType {
+    File,
+    Symlink,
+}
+
+enum DateType {
+    NotToday,
+    Today,
+}
+
+enum YearType {
+    NotThisYear,
+    CurrentYear,
+}
+
+enum BinarySizeUnit {
+    Bytes,
+    Kibibytes,
+    Mebibytes,
+    Gibibytes,
+    Tebibytes,
+    Pebibytes,
+    Exbibytes,
+    Zebibytes,
+    Yobibytes,
+}
```

Comment:
Let's just handle up to gigabytes.

Reply:
Done. Latest changes have now been committed to this pull-request branch.